### PR TITLE
fix: check if requestcontext exists

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: src/main/java/nva/commons/utils/MockClaims.java [hunk 1]
```diff
@@ -15,17 +15,23 @@
...
+    public static JsonNode apiGatewayEvent(JsonNode event, ObjectMapper jsonParser) {
+        ObjectNode copy = event.deepCopy();
+
+        if (!copy.has(REQUEST_CONTEXT_ROOT_NODE)) {
+            copy.set(REQUEST_CONTEXT_ROOT_NODE, jsonParser.createObjectNode());
+        }
+
         ObjectNode requestContext = (ObjectNode) copy.get(REQUEST_CONTEXT_ROOT_NODE);
-        requestContext.set(AUTHORIZER_NODE, objectMapper.createObjectNode());
+        requestContext.set(AUTHORIZER_NODE, jsonParser.createObjectNode());
```
Comment (janbj) [inline src/main/java/nva/commons/utils/MockClaims.java:32]: Don't like the name objectMapper?

File: src/main/java/nva/commons/utils/MockClaims.java [hunk 2]
```diff
@@ -15,17 +15,23 @@
...
+     * @param event      the ApiGateway event
+     * @param jsonParser a jsonParser
      * @return an event with mocked claims
      */
-    public static JsonNode apiGatewayEvent(JsonNode event, ObjectMapper objectMapper) {
-        JsonNode copy = event.deepCopy();
+    public static JsonNode apiGatewayEvent(JsonNode event, ObjectMapper jsonParser) {
+        ObjectNode copy = event.deepCopy();
+
+        if (!copy.has(REQUEST_CONTEXT_ROOT_NODE)) {
```
Comment (janbj) [inline src/main/java/nva/commons/utils/MockClaims.java:27]: When can it happen that we have no root node?

Comment (axthosarouris) [reply src/main/java/nva/commons/utils/MockClaims.java:27]: Maybe the constant name is wrong. It is not the root node. It the requestContext node
The request sent from SAM (local run):

```
{
  "headers": {
    "Authorization": "some api key",
    "Content-Type": "application/json"
  },
  "body": "{\n  \"createdDate\": \"2020-01-29T14:32:43.770Z\",\n  \"status\": \"New\",\n  \"modifiedDate\": \"2020-01-29T14:32:43.770Z\",\n  \"owner\": \"\",\n  \"identifier\": \"4d96e658-c2e0-4f23-9f1d-ccae0c770ecd\",\n  \"entityDescription\": {\n    \"type\": \"JournalArticle\",\n    \"mainTitle\": \"Toward unique identifiers\",\n    \"date\": {\n      \"year\": \"1999\"\n    },\n    \"contributors\": [\n      {\n        \"name\": \"Paskin, N.\",\n        \"nameType\": \"Personal\",\n        \"sequence\": 0\n      }\n    ]\n  }\n}"
}
```

The node requestContext

File: src/main/java/nva/commons/utils/MockClaims.java (hunk 1 above)
Comment (axthosarouris) [reply src/main/java/nva/commons/utils/MockClaims.java:32]: I have always to remember how it is cold. When I am doing json parsing i want to write json and the variable to appear :)

Comment (janbj) [reply src/main/java/nva/commons/utils/MockClaims.java:32]: It's not just a parser you know :)

**Pull request merged**